master/reference/advanced/etcd-rbac/certificate-generation.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Generating certificates with usernames for etcd RBAC
+---
+
+Generating Certificate Authority(CA), Certificates, and Keys can be
+accomplished with many different tools.  This document tries to layout the
+unique or specific details that are needed for each of the different
+certificates but uses the
+[hack/tls-setup tool from the etcd repo](https://github.com/coreos/etcd/tree/master/hack/tls-setup),
+to make certificate generation easy.  If you use hack/tls-setup too, ensure
+you have followed the README and are able to run `make` successfully.
+
+It is important to ensure that the Common Name (CN), of each component that
+will be accessing etcd, match the username in etcd that has the appropriate
+etcd roles for accessing its needed keys or paths.
+
+> **Note:** Re-running `make` using hack/tls-setup will regenerate the CA and
```

Comment:
This note feels out of place here, perhaps move to the overview of running the script.